eng/scripts/live-test-resource-cleanup.ps1

```diff
@@ -35,61 +47,226 @@ param (
     $IgnoreUnusedArguments
 )
 
+Set-StrictMode -Version 3
+
 &"$PSScriptRoot/../common/scripts/Import-AzModules.ps1"
 
 # Import resource management helpers and override its Log function.
-. "$PSScriptRoot\..\common\scripts\Helpers\Resource-Helpers.ps1"
+. "$PSScriptRoot/../common/scripts/Helpers/Resource-Helpers.ps1"
+
+$OwnerAliasCache = @{}
 
 function Log($Message) {
   Write-Host $Message
 }
 
+function IsValidAlias
+{
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Alias
+    )
+
+    if ($OwnerAliasCache.ContainsKey($Alias)) {
+      return $OwnerAliasCache[$Alias]
+    }
+
+    $domains = @("microsoft.com", "ntdev.microsoft.com")
+
+    foreach ($domain in $domains) {
+        if (Get-AzAdUser -UserPrincipalName "$Alias@$domain") {
+            $OwnerAliasCache[$Alias] = $true
+            return $true;
+        }
+    }
+
+    $OwnerAliasCache[$Alias] = $false
 
-Write-Verbose "Logging in"
-$provisionerSecret = ConvertTo-SecureString -String $ProvisionerApplicationSecret -AsPlainText -Force
-$provisionerCredential = [System.Management.Automation.PSCredential]::new($ProvisionerApplicationId, $provisionerSecret)
-Connect-AzAccount -Force -Tenant $TenantId -Credential $provisionerCredential -ServicePrincipal -Environment $Environment
-Select-AzSubscription -Subscription $SubscriptionId
+    return $false;
+}
 
-Write-Verbose "Fetching groups"
-$allGroups = @(Get-AzResourceGroup)
+function GetTag([object]$ResourceGroup, [string]$Key) {
+  if (!$ResourceGroup.Tags) {
+    return $null
+  }
 
-Write-Host "Total Resource Groups: $($allGroups.Count)"
+  foreach ($tagKey in $ResourceGroup.Tags.Keys) {
+    # Compare case-insensitive
+    if ($tagKey -like $Key) {
```

Comment:
Why like operator here instead of equals?

Reply:
By default eq is case insensitive https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_comparison_operators?view=powershell-7.2#common-features

Reply:
🤯

Reply:
And if you ever want to force case-sensitivity or insensitivity, put "c" before case-sensitive and "i" before case-insensitive operators (when applicable).